src/tunnel.rs

```diff
@@ -77,7 +77,12 @@ impl<S: NetSession> Tunnel<S> {
             if count > 0 {
                 return Err(io::ErrorKind::TimedOut.into());
             }
-            while let Some((fd, ev)) = poller.next() {
+            while let Some((fd, res)) = poller.next() {
+                let ev = match res {
+                    Ok(ev) => ev,
+                    Err(IoFail::Connectivity(_)) => return Ok((in_count, out_count)),
```

Comment:
I think this should be an error, as it's not a graceful shutdown

Reply:
When we get HANGUP this `IoFail` is returned, so we assume the tunnel is closed gracefully. How else to detect that the stream is over?